_drafts/PR-blogs/planning-docs/example-code/animated-dots/original/script.js

```diff
@@ -0,0 +1,170 @@
+window.onload = init();
+function init() {
+  canvas = document.getElementById("canvas");
+  context = canvas.getContext("2d");
+  canvas.height = window.innerHeight;
+  canvas.width = window.innerWidth;
+  mouse = { x: 0, y: 0 };
+  colors = [
```

Comment:
Let's define these pieces of global state (e.g.: `mouse`, `colors`, `dotsHolder`, etc.) as global variables within the global scope of the file.

```js
// not this:
function init() {
  globalMutableVariable = "hi mom" 
}

console.log(window.globalMutableVariable) // -> "hi mom"

// instead, this:
let globalMutableVariable = "hi mom" 

console.log(globalMutableVariable) // -> "hi mom"
```

Defining these variables using `let`, instead of implicitly defining them as global variables on the `window` object, like you're currently doing, doesn't really change anything functionally. But it does make your code more modern and idiomatic, and relieves you of the need to define these globals within the `init` function, which will further separate out app state and app code from bootstrapping code.